internal/mode/static/nginx/modules/src/httpmatches.js

```diff
@@ -1,16 +1,15 @@
-import qs from 'querystring';
-
-const MATCHES_VARIABLE = 'http_matches';
+const MATCHES_KEY = 'match_key';
 const HTTP_CODES = {
   notFound: 404,
   internalServerError: 500,
 };
 
 function redirect(r) {
-  let matches;
+  let matchList;
 
   try {
-    matches = extractMatchesFromRequest(r);
+    let key = r.variables[MATCHES_KEY];
+    matchList = matches[key];
```

Comment:
So `matches` variable is set by `js_preload_object`? We don't have to declare it here?

Reply:
we are not setting `matches` variable. You are right that `js_preload_object` maps the `matches` object but `matches` is a map of key-value for http match pairs. So, we are fetching the http matches from using the key. We only know the value of matches at runtime, which in our case is a curl request

Does this make sense?

Reply:
Got it, thanks